core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -215,7 +216,11 @@ public void writeLong(long i) throws IOException {
      * Negative numbers are not supported.
      */
     public void writeVLong(long i) throws IOException {
-        assert i >= 0;
+        if (Assertions.ENABLED) {
```

Comment:
Yeah, that's fine with me. I just needed this to fail with something that would get caught and sent back in the response instead of uncaught causing the request to timeout. I'll take a look at #22314 now.